docs/frontend_architecture/unit_testing.rst

```diff
@@ -21,4 +21,79 @@ Tests are written in JavaScript, and placed in the 'assets/test' folder. An exam
   });
 
 
-Vue.js components can also be tested. The management plugin contains an example (*kolibri/plugins/management/assets/test/management.js*) where the component is bound to a temporary DOM node, changes are made to the state, and assertions are made about the new component structure.
+For the frontend testing of ``Vue.js`` components, we make use of ```Vue Testing Library`` <https://testing-library.com/docs/vue-testing-library/intro/>`__
+and the associated ecosystem.
+
+It is based on the philosphy that “The more your tests resemble the way your software is used, the more confidence they can give you." Rather than dealing with instances of rendered Vue components, it allows our tests to work with actual DOM nodes in the same way the user would. This deals to more user centric and a better quality of tests generally, and thus we have been making efforts to use the same (in constrast to the earlier used ```Vue Test Utils`` <https://v1.test-utils.vuejs.org/>`__).
+
+If you have never worked with Vue Testing Library's (VTL) before, it is highly recommeed to the `VTL examples page <https://testing-library.com/docs/vue-testing-library/examples>`__ to see the library in action and understand it's use. If you're comfortable with the same, you can either:
+
+-  Read some of the exisiting tests in Kolibri to see how we make use of the same.
+-  Checkout the `test template <TODO>`__ and the `style guide <TODO>`__ to start writing your own!
```

Comment:
I am unsure as to how to add an internal link and external link to documentation on an `.rst` file. Would appreciate any help with the same.

Reply:
Thanks!